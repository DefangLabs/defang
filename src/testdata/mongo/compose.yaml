# Use root/example as user/password credentials

services:
  mongo:
    image: mongo
    restart: always
    environment:
      MONGO_INITDB_ROOT_USERNAME: root
      MONGO_INITDB_ROOT_PASSWORD: example!
    x-defang-mongodb: true

  mongo-express:
    image: mongo-express
    restart: always
    ports:
      - 8081:8081
    environment:
      ME_CONFIG_MONGODB_ADMINUSERNAME: root
      ME_CONFIG_MONGODB_ADMINPASSWORD: example!
      ME_CONFIG_MONGODB_URL: mongodb://root:example!@mongo:27017/
      ME_CONFIG_BASICAUTH: false
    depends_on:
      - mongo
