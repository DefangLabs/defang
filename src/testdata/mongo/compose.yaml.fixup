{
  "mongo": {
    "command": null,
    "entrypoint": null,
    "environment": {
      "MONGO_INITDB_ROOT_PASSWORD": "example!",
      "MONGO_INITDB_ROOT_USERNAME": "root"
    },
    "image": "mongo",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 27017,
        "protocol": "tcp"
      }
    ],
    "restart": "always"
  },
  "mongo-express": {
    "command": null,
    "depends_on": {
      "mongo": {
        "condition": "service_started",
        "required": true
      }
    },
    "entrypoint": null,
    "environment": {
      "ME_CONFIG_BASICAUTH": "false",
      "ME_CONFIG_MONGODB_ADMINPASSWORD": "example!",
      "ME_CONFIG_MONGODB_ADMINUSERNAME": "root",
      "ME_CONFIG_MONGODB_URL": "mongodb://root:example!@mongo:27017/"
    },
    "image": "mongo-express",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 8081,
        "published": "8081",
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ],
    "restart": "always"
  }
}