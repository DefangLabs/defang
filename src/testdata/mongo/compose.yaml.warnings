 ! service "mongo": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "mongo": stateful service will lose data on restart; use a managed service instead
 ! service "mongo-express": environment "ME_CONFIG_MONGODB_URL" may contain sensitive information; consider using 'defang config set ME_CONFIG_MONGODB_URL' to securely store this value
 ! service "mongo-express": ingress port without healthcheck defaults to GET / HTTP/1.1
 ! service "mongo-express": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "mongo-express": service name was adjusted: environment variable "ME_CONFIG_MONGODB_URL" assigned value "mongodb://root:example!@mock-mongo:27017/"
