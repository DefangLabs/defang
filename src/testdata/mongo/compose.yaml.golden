name: mongo
services:
  mongo:
    environment:
      MONGO_INITDB_ROOT_PASSWORD: example!
      MONGO_INITDB_ROOT_USERNAME: root
    image: mongo
    networks:
      default: null
    restart: always
    x-defang-mongodb: true
  mongo-express:
    depends_on:
      mongo:
        condition: service_started
        required: true
    environment:
      ME_CONFIG_BASICAUTH: "false"
      ME_CONFIG_MONGODB_ADMINPASSWORD: example!
      ME_CONFIG_MONGODB_ADMINUSERNAME: root
      ME_CONFIG_MONGODB_URL: mongodb://root:example!@mongo:27017/
    image: mongo-express
    networks:
      default: null
    ports:
      - mode: ingress
        target: 8081
        published: "8081"
        protocol: tcp
    restart: always
networks:
  default:
    name: mongo_default
